### PR TITLE
chore(ci) use testcontainers for postgres tests

### PR DESCRIPTION
**Pull request by lahabana**
### Summary

Remove the specific build flags for postgres tests and
simply use test-containers.
This makes a single test target and removes some scripts.

### Backwards compatibility

- [x] Add `backport-to-stable` label if the code is backwards compatible. Otherwise, list breaking changes.

### CODE REVIEW

File: pkg/test/store/postgres/test_container.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	"github.com/kumahq/kuma/pkg/config"
+	pg_config "github.com/kumahq/kuma/pkg/config/plugins/resources/postgres"
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/plugins/common/postgres"
+)
+
+type PostgresContainer struct {
+	container testcontainers.Container
+	WithSsl   bool
```
Comment (jpeach) [inline pkg/test/store/postgres/test_container.go:21]:
```suggestion
	WithTLS   bool
```

File: pkg/test/store/postgres/test_container.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func resourceDir() string {
+	dir, _ := os.Getwd()
+	for path.Base(dir) != "pkg" && path.Base(dir) != "app" {
+		dir = path.Dir(dir)
+	}
```
Comment (jpeach) [inline pkg/test/store/postgres/test_container.go:28]: err, what?

File: pkg/test/store/postgres/test_container.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+func (v *PostgresContainer) Stop() error {
+	if v.container != nil {
+		return v.container.Terminate(context.Background())
+	}
+	return nil
+}
+
+func (v *PostgresContainer) Config(withRandomDB bool) (*pg_config.PostgresStoreConfig, error) {
```
Comment (jpeach) [inline pkg/test/store/postgres/test_container.go:72]: Suggest using constants `WithRandomDB` and `WithoutRandomDB` to make the call sites more readable.

File: pkg/plugins/leader/postgres/leader_elector_test.go [hunk 4]
```diff
@@ -11,39 +8,22 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
-	"github.com/kumahq/kuma/pkg/config"
-	"github.com/kumahq/kuma/pkg/config/plugins/resources/postgres"
 	"github.com/kumahq/kuma/pkg/core/runtime/component"
 	common_postgres "github.com/kumahq/kuma/pkg/plugins/common/postgres"
 	leader_postgres "github.com/kumahq/kuma/pkg/plugins/leader/postgres"
 	"github.com/kumahq/kuma/pkg/test"
+	pg_test "github.com/kumahq/kuma/pkg/test/store/postgres"
```
Comment (jpeach) [inline pkg/plugins/leader/postgres/leader_elector_test.go:15]: nit: looks like the import alias convention would be `test_postgres`?

**Review by jpeach — APPROVED** [review]
Nice simplification!

File: pkg/plugins/leader/postgres/leader_elector_test.go (hunk 4 above)
Comment (lahabana) [reply pkg/plugins/leader/postgres/leader_elector_test.go:15]: Is it something quite systematic to have these types of aliases?

File: pkg/test/store/postgres/test_container.go (hunk 2 above)
Comment (lahabana) [reply pkg/test/store/postgres/test_container.go:28]: It's finding the root directory of the project. Do you have another way of doing this?

File: pkg/test/store/postgres/test_container.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+		Env: map[string]string{
+			"POSTGRES_USER":     "kuma",
+			"POSTGRES_PASSWORD": "kuma",
+			"POSTGRES_DB":       "kuma",
+		},
+		ExposedPorts: []string{"5432/tcp"},
+		WaitingFor:   wait.ForListeningPort("5432"),
+	}
+	if v.WithSsl {
+		req.Cmd = strings.Split("-c ssl=on -c ssl_cert_file=/var/lib/postgresql/postgres.server.crt -c ssl_key_file=/var/lib/postgresql/postgres.server.key -c ssl_ca_file=/var/lib/postgresql/rootCA.crt -c hba_file=/var/lib/postgresql/pg_hba.conf", " ")
```
[the remark this answers is not in the document]
Comment (lahabana) [reply pkg/test/store/postgres/test_container.go:52]: I copy pasted the line as is from somewhere else. I guess laziness would be an answer to your question :)

**Review by jakubdyszkiewicz — CHANGES_REQUESTED** [review]
Does `make start/postgres`, `make start/postgres/ssl` still work? It is needed to run Kuma with Postgres `make run/universal/postgres` for local development

**Comment by lahabana** [comment]
@jakubdyszkiewicz fixed so that the make targets still work.

**Review by jakubdyszkiewicz — APPROVED** [review]
thanks

**Pull request merged**